src/Compilers/CSharp/Portable/Symbols/Compilation_WellKnownMembers.cs

```diff
@@ -97,8 +97,18 @@ internal NamedTypeSymbol GetWellKnownType(WellKnownType type)
                 }
                 else
                 {
-                    result = this.Assembly.GetTypeByMetadataName(
-                        mdName, includeReferences: true, useCLSCompliantNameArityEncoding: true, isWellKnownType: true, warnings: warnings);
+                    if (type <= WellKnownType.CSharp7Sentinel)
```

Comment:
Consider using conditional on `warnings` parameter in call: `warnings: (type < Sentinel) ? null : warnings`.